Format Time correctly with Formatter

mathias-lang-sociomantic opened the pull request:
CC @andrej-mitrovic-sociomantic : Do you need other format string than 'G' ?

Fixes #33

mathias-lang-sociomantic committed:
Remove versioned-out code

mathias-lang-sociomantic committed:
Modify ocean.text.convert.DateTime_tango to work with sink

Currently formatCustom can format into a buffer.
However the formatter passes a `Sink` to the `toString` method, to avoid needless allocations.
The changes to use a sink are actually trivial, as it's a superset of buffers.

gavin-norman-sociomantic commented on src/ocean/text/convert/DateTime_tango.d:
```diff
@@ -586,11 +604,9 @@ struct DateTimeLocale
-     **********************************************************************/
-
-    private char[] formatCustom (ref Result result, Time dateTime, cstring format)
+    /// Implementation of format
```
Would be nice to add doc in the standard format, if you're going to add it. (Despite this module's doc being very non-standard ;)

mathias-lang-sociomantic replied:
I'd rather keep any cleanup for another day (I prefer to do all at once). Will revert :)

gavin-norman-sociomantic commented on src/ocean/text/convert/DateTime_tango.d:
```diff
@@ -586,11 +604,9 @@ struct DateTimeLocale
-
-    private char[] formatCustom (ref Result result, Time dateTime, cstring format)
+    /// Implementation of format
+    private void formatCustom (size_t delegate(cstring) sink, Time dateTime,
```
Suggest splitting this change to `formatCustom` to accept a sink delegate into a separate commit. It's not clear to me where the "Format Time correctly with Formatter" part comes in.

mathias-lang-sociomantic committed:
Format Time correctly with Formatter

Before a Time structure was formatted as any struct (printing the only member, ticks_).
With this change we get a behaviour closer to the one of Tango_Layout, as it formats it as a human-readable string.
However, the formatting is currently limited to the 'G' formatting, as we lack a way to transmit format specifier to 'toString'

andrej-mitrovic-sociomantic commented:
> Do you need other format string than 'G' ?

I only need whatever dolphin's output needs. I don't recall which format string it is though.

andrej-mitrovic-sociomantic commented:
I'm not sure what `<redacted>` is using. But if the tests of `<redacted>` pass, we're good to go, stranger.

mathias-lang-sociomantic commented:
Updated:
- Reverted the minor doc change
- Split into 2 commits.

mathias-lang-sociomantic commented:
Also rebased on v2.5.x

andrej-mitrovic-sociomantic commented:
You have a typo in the commit message: `speficier`

mathias-lang-sociomantic commented:
Made the message less `speficic`

mathias-lang-sociomantic commented:
Ping

mihails-strasuns-sociomantic commented:
@andrej-mitrovic-sociomantic can you please confirm that this works for you?

andrej-mitrovic-sociomantic commented:
@mathias-baumann-sociomantic can you please confirm that it works for me?

andrej-mitrovic-sociomantic commented:
wrong mathias

andrej-mitrovic-sociomantic commented:
@mathias-lang-sociomantic ^

mathias-lang-sociomantic commented:
It does

mihails-strasuns-sociomantic commented:
Admirable teamwork :)

mihails-strasuns-sociomantic commented:
@mathias-lang-sociomantic if this goes as a feature, should also do release notes mention

mihails-strasuns-sociomantic commented:
Marking as a bug fix after chat with @mathias-lang-sociomantic
